internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -178,17 +180,23 @@ func (ws *workflows) sessionWorkflow(wctx workflow.Context, params *sessionWorkf
 	//                blow us up due to non determinism.
 
 	var prints []string
+	// attrs := map[string]string{"session_id": sessionID}
 
 	data, err := ws.getSessionData(wctx, params.SessionID)
 	if err == nil {
+		// FIXME: session_id? environment?
+		telemetry.Metrics.Sessions.Update(1, map[string]string{"session_id": sessionID, "type": "start"})
```

Comment:
I think a better way here is to harden the possibilities

telemetry should expose a simple interface with all our possible metrics and each component should just call a simple method 

```go
telemetry.IncSessionCount()
telemetry.IncRunningSessionCount()
telemetry.DecRunningSessionCount()
```

in this way, we have one interface for all metrics we expose
we also control the cardinality of the dimensions 
and we make stuff less error prone and more readable

it can be multiple interfaces for each component instead of one that does everything 

we can expose the underlying sdk if in some cases we need a raw level access (like db interface vs gorm vs raw sql query)

Reply:
that's actually what I did, IMO
- you have metrcis file in which you define metric and it's type. Very simple.
- I wrap opentelemetry metrics and methods  to provide one simple interface and method - e.g. Update(). All other things passed are optional tags.
- Specificaly sessions defined as UpDOwnCounter which is similar to Gauge. You could find through it both overall sessions count per env, amount/ratio of failed/stopped and even duration of each session.

What you are suggesting - if I understand you right - requires much more work for developer while adding new metric. In my version, it requires literally nothing - just add name and type to the metric file.

Reply:
You can provide an internal implementation that every new metric can use (like this update for example), but the telemetry api should expose a very opinionated and strict api.

So developers can add metrics and use metrics easily.

Reply:
agree - that's why I think what I suggest is the simplest version.
We are mimicing opentelemetry and even simplifying it more.
Open telemetry API have Add() method for Counter, another Add method for UpDownCounter, Record() for histogram etc. Having one Update() method is simpler and clearer

Reply:
`telemetry.Metrics.Sessions.Update(-1)` is really not clear. Either do what @efiShtain suggested, or imho better yet just expose the actual metric object from otel. Adding another abstraction layer here just complicates stuff.

Reply:
@itayd - please see metrics RFC.
- I don't want to expose opentelemetry objects directly. I think we need a simple abstraction layer in order to be able to replace/adjust/improve it later
- It's not easy to pass context tags in open telemetry. This also need to be abtracted
- Open telemetry Counters' update method is `Add`. And Gaude's update method is `Set`. My Idea was to introduce  single API to all metrics, but I could change methods naming according to openTelemetry one